### PR TITLE
add chain of trust docs, from the worker perspective.

### LINKED ISSUE #1140: add worker-oriented chain of trust documentation
We have chain of trust documentation [here](https://scriptworker.readthedocs.io/en/latest/chain_of_trust.html) and a roadmap [here](https://docs.google.com/document/d/1nr8mBpiK527ojD1yeTJYONtDRE3ADBCbIWm7q8nRgSA/edit), but we don't have any documentation for the docker-worker and generic-worker implementation and maintenance.

### PR DESCRIPTION
**Pull request by escapewindow**
Fixes #1140.

### CODE REVIEW

**Comment by djmitche** [comment]
GitHub should have an "approved with revisions" button.  If you make a quick fix to the points above I'll get this landed.

File: dev-docs/chain-of-trust.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    - We have included `publicIpAddress`, `privateIpAddress`, `instanceId`, `instanceType`, and `region` to docker-worker and generic-worker environment metadata, as metadata that could be helpful in auditing or debugging issues. There is no strict schema here at the moment, should the set of useful metadata change.
+
+We currently prefer if the `chain-of-trust.json` artifact is indented for easier human readability. (We also previously hit a gpg line length limit issue, but we no longer use gpg to sign the CoT artifact.)
+
+### Chain of Trust signature
+
+In a subset of cases, we generate a detached ed25519 signature of the Chain of Trust artifact. This signature is uploaded as `public/chain-of-trust.json.sig` .
+
+In our development and CI worker pools, we ignore signature verification, so we don't need to generate the signature, though it isn't harmful to do so. In our trusted release worker pools, we require that the Chain of Trust artifact is signed by a valid ed25519 key.
```
[the remark this answers is not in the document]
Comment (escapewindow) [reply dev-docs/chain-of-trust.md]: Hm.

I made an attempt here, lmk what you think?

**Comment by escapewindow** [comment]
> I like this as a first step. I think this could form the basis of a page in the "[Conventions](https://docs.taskcluster.net/docs/manual/design/conventions)" section of the manual, along with copying the schema into the TC schemas. We could then add a section to "[Using Taskcluster](https://docs.taskcluster.net/docs/manual/using)" around ways to deploy and use cot artifacts.

Are these future followup ideas, or part of what I should be doing pre-merge?

File: dev-docs/chain-of-trust.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+#### Security of the private key
+
+Because the ed25519 private key acts as our second factor, we need to safeguard the security of the private key.
+
+- This is our second factor to help secure Taskcluster scopes. Therefore, maintainers should not be able to access this secret without some credentials other than Taskcluster scopes.
+- The file should be read-only, and only readable by the worker user.
+- The task user must not be able to read the file.
+- By limiting the release key to release-only workerTypes, and by limiting which users, repos, and tasks have scopes to launch release-only workerTypes, we limit which tasks can run on those pools.
```
[the remark this answers is not in the document]
Comment (escapewindow) [reply dev-docs/chain-of-trust.md]: Adjusted.

**Comment by djmitche** [comment]
Those are future followup :)

File: dev-docs/chain-of-trust.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (djmitche) [reply dev-docs/chain-of-trust.md]: Thanks!

**Pull request merged**